pyroute2/ipset.py

```diff
@@ -243,6 +243,10 @@ def _family_to_version(family):
 
     def _entry_to_data_attrs(self, entry, etype, ip_version):
         attrs = []
+
+        # Set the default protocol to tcp for hash:ports
+        p = lambda x: socket.getprotobyname('tcp') if x == None else x
```

Comment:
could it be solved simply by default values of `PortRange.protocol` and `PortEntry.protocol` ?

Reply:
No, since several ipset modules don't support protocol options. And I'm not sure of support of all modules revisions of hash:ports, by the way.

Personally, I'm against setting default values here. They are too many modules/revisions combinations, we will probably break something. That should be done in upper layers, if someone need it.

Reply:
Ok, reasonable.

I'm just thinking how to avoid lambda assignment here.